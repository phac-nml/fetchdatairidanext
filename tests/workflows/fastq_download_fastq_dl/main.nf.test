nextflow_workflow {

    name "Test workflow: fastq_download_fastq_dl/main.nf"
    script "subworkflows/local/fastq_download_fastq_dl/main.nf"
    workflow "FASTQ_DOWNLOAD_FASTQ_DL"
    tag "subworkflow"
    tag "fastq_dl"
    tag "fastq_download_fastq_dl"

    test("Parameters: default") {

        when {
            workflow {
                """
                input[0] = Channel.of(
                    [[ irida_id:'test_single_end', single_end:true  ], 'DRR000774'],
                    [[ irida_id:'test_paired_end', single_end:false ], 'SRR11140744']
                )
                """
            }
            params {
                outdir = "output"
                provider = "ENA"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )

            // Output files
            assert path("$launchDir/output").exists()
            assert path("$launchDir/output/reads/DRR000774.fastq.gz").md5 == "2fa2a3ef2dccd1799278097860a8c031"
            assert path("$launchDir/output/reads/SRR11140744_1.fastq.gz").md5 == "a31c985a614b05e8beb05da7b6ea9c6d"
            assert path("$launchDir/output/reads/SRR11140744_2.fastq.gz").md5 == "353d40a9da709338ee7cf1f5e694c0c8"

            // Failures is empty
            def lines = path("$launchDir/output/prefetch/failures_report.csv").readLines()
            assert lines.size() == 1
            assert lines.contains("sample,error_accession")
            assert lines.contains("test_single_end,DRR000774").equals(false)
            assert lines.contains("test_paired_end,SRR11140744").equals(false)
        }
    }

    test("Download errors: 403 and invalid") {

        when {
            workflow {
                """
                input[0] = Channel.of(
                    [[ irida_id:'SAMPLE1', single_end:false ], 'ERR1109373'],
                    [[ irida_id:'ERROR1', single_end:false ], 'SRR999908'],
                    [[ irida_id:'ERROR2', single_end:false ], 'INVALID!!'],
                    [[ irida_id:'SAMPLE2', single_end:false ], 'SRR13191702']
                )
                """
            }
            params {
                outdir = "output"
                provider = "ENA"
            }
        }

        then {
            assert workflow.success
            assert path("$launchDir/output").exists()
            assert path("$launchDir/output/reads/ERR1109373_1.fastq.gz").md5 == "4cb0072f56d92847cb5d33edd78e1b00"
            assert path("$launchDir/output/reads/ERR1109373_2.fastq.gz").md5 == "ac356dfe2caae9b30d4598eb954cd340"
            assert path("$launchDir/output/reads/SRR13191702_1.fastq.gz").md5 == "baaaea61cba4294ec696fdfea1610848"
            assert path("$launchDir/output/reads/SRR13191702_2.fastq.gz").md5 == "8e43ad99049fabb6526a4b846da01c32"

            def lines = path("$launchDir/output/prefetch/failures_report.csv").readLines()
            assert lines.size() == 3
            assert lines.contains("sample,error_accession")
            assert lines.contains("ERROR1,SRR999908")
            assert lines.contains("ERROR2,INVALID!!")
        }
    }
}
