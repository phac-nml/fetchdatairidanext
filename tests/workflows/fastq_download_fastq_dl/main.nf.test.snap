{
    "Parameters: default": {
        "content": [
            {
                "0": [
                    [
                        {
                            "irida_id": "test_paired_end",
                            "single_end": false
                        },
                        [
                            "SRR11140744.fastq.gz:md5,2adaca2017c2710576adc31e612b4e0c",
                            "SRR11140744_1.fastq.gz:md5,a31c985a614b05e8beb05da7b6ea9c6d",
                            "SRR11140744_2.fastq.gz:md5,353d40a9da709338ee7cf1f5e694c0c8"
                        ]
                    ],
                    [
                        {
                            "irida_id": "test_single_end",
                            "single_end": true
                        },
                        "DRR000774.fastq.gz:md5,2fa2a3ef2dccd1799278097860a8c031"
                    ]
                ],
                "1": [
                    "versions.yml:md5,6370b8b2476b5bc54b7001c33197b577"
                ],
                "reads": [
                    [
                        {
                            "irida_id": "test_paired_end",
                            "single_end": false
                        },
                        [
                            "SRR11140744.fastq.gz:md5,2adaca2017c2710576adc31e612b4e0c",
                            "SRR11140744_1.fastq.gz:md5,a31c985a614b05e8beb05da7b6ea9c6d",
                            "SRR11140744_2.fastq.gz:md5,353d40a9da709338ee7cf1f5e694c0c8"
                        ]
                    ],
                    [
                        {
                            "irida_id": "test_single_end",
                            "single_end": true
                        },
                        "DRR000774.fastq.gz:md5,2fa2a3ef2dccd1799278097860a8c031"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6370b8b2476b5bc54b7001c33197b577"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-03T10:31:24.480521123"
    }
}